Use wxGtkValue RAII wrapper in wxGTK code

vadz opened the pull request:
wxGtkValue class was added back in 3f84cb17ca (Add wxActivityIndicator
control., 2015-03-06), but somehow never used. Start using it now
(better late than never...) as it makes the code simpler, shorter and
more robust.

No real changes.

vadz committed:
Use wxGtkValue RAII wrapper in wxGTK code

wxGtkValue class was added back in 3f84cb17ca (Add wxActivityIndicator
control., 2015-03-06), but somehow never used. Start using it now
(better late than never...) as it makes the code simpler, shorter and
more robust.

No real changes.

paulcor commented on include/wx/gtk/private/value.h:
```diff
@@ -17,6 +17,11 @@
 {
 public:
+    explicit wxGtkValue()
+        : m_val(G_VALUE_INIT)
```
`G_VALUE_INIT` is a braced-init-list, using it like this requires C++11. Same for the ctor below.

vadz replied:
Thanks for noticing this. AFAICS the only C++98-compatible way to fix this is to use `memset(&m_val, 0, sizeof(m_val));`, do you see anything less clumsy?

paulcor replied:
That's the only way I'm aware of.

vadz committed:
Initialize wxGtkValue in a way not requiring C++11

As we still support C++98, don't use G_VALUE_INIT as the initializer:
it's a braced-init-list, which is only allowed in C++11.

Co-Authored-By: Paul Cornett <paulcor@users.noreply.github.com>
